src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -1338,19 +1439,189 @@ def main():
             start_time = time.time()
 
         build_tests.parity_check(
-            parity_parameters, run_parameters, results,
+            parity_set,
+            run_results,
         )
 
         if verbose:
             print("parity check complete")
         if showtiming:
             print("... in %s seconds." % (time.time() - start_time))
 
+
+def new_nwm_q0(run_results):
+    return pd.concat(
+        # TODO: we only need two fields, technically, and the restart file produced by WRF-Hydro
+        # actually contains a field qu0, which is never used for restart (the qu0 can be obtained
+        # as the qd0 from the topologically upstream segments, just like during the calculation).
+        # In any case, the qu0 currently in the WRF-Hydro output is populated with the same value
+        # as the qd0.
+        #[pd.DataFrame(d[:,-3::2], index=i, columns=["qd0", "h0"]) for i, d in run_results],
+        #[pd.DataFrame(r[1][:,-3:], index=r[0], columns=["qu0", "v0", "h0"]) for r in run_results],
+        [pd.DataFrame(r[1][:,[-3,-3,-1]], index=r[0], columns=["qu0", "qd0", "h0"]) for r in run_results],
+        copy=False,
+    )
+
+
+if __name__ == "__main__":
+
+    (
+        log_parameters,
+        supernetwork_parameters,
+        waterbody_parameters,
+        compute_parameters,
+        forcing_parameters,
+        restart_parameters,
+        output_parameters,
+        parity_parameters,
+        data_assimilation_parameters,
+        diffusive_parameters,
+    ) = _input_handler()
+
+    verbose = log_parameters.get("verbose", None)
+    showtiming = log_parameters.get("showtiming", None)
+    debuglevel = log_parameters.get("debuglevel", 0)
+
+    if showtiming:
+        main_start_time = time.time()
+
+    (
+        connections,
+        param_df,
+        wbodies,
+        waterbodies_df,
+        break_network_at_waterbodies,
+        independent_networks,
+        reaches_bytw,
+        rconn,
+    ) = nwm_network_preprocess(
+        supernetwork_parameters,
+        waterbody_parameters,
+        showtiming=showtiming,
+        verbose=verbose,
+        debuglevel=debuglevel,
+    )
+
+    waterbodies_df, q0 = nwm_initial_warmstate_preprocess(
```

Comment:
Remember: we need to set this up so that the return is just the initial states from the waterbodies (rather than a populated dataframe) That will have to be adjusted downstream to make sure the compute_network function (and other items) are looking for the additional argument.